### ISSUE TITLE
npm test fails for js-ipfs-api with TypeError

### ISSUE THREAD

**Issue by wigy-opensource-developer** [haadcode/orbit-db#141]
This is the relevant part of the test output, the same error is repeated 25 times for all js-ipfs-api tests but 2:

```
  108 passing (41s)
  25 failing

  1) Orbit Client with js-ipfs-api Add events adds an item to an empty channel:
     TypeError: callback is not a function
      at size (node_modules/ipld-dag-pb/src/dag-node.js:210:9)
      at util.serialize (node_modules/ipld-dag-pb/src/dag-node.js:183:7)
      at Object.exports.serialize.err [as serialize] (node_modules/ipld-dag-pb/src/util.js:55:3)
      at DAGNode.size (node_modules/ipld-dag-pb/src/dag-node.js:170:10)
      at multihash (node_modules/ipld-dag-pb/src/dag-node.js:198:12)
      at DAGNode.multihash (node_modules/ipld-dag-pb/src/dag-node.js:161:7)
      at DAGNode.toJSON (node_modules/ipld-dag-pb/src/dag-node.js:194:10)
      at ipfs.object.put.then (node_modules/ipfs-log/src/entry.js:37:26)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)
```

Looking at the problem, it seems that `DAGNode.toJSON` has a "nodeback" API since [ipld-dag-pb 0.8.0](https://github.com/ipld/js-ipld-dag-pb/commit/d2bf30373e59a0c66ba5f834da5723a6bac798ef), but [toIpfsHash in entry.js of ipfs-log 1.5.2](https://github.com/haadcode/ipfs-log/blob/master/src/entry.js#L37) still calls it synchronously without a callback.

Because of dependencies on the master branch of js-ipfs-api, it might be difficult to reproduce the same environment on your machine to reproduce this issue.

**Comment by haadcode** [comment]
Hi @wigy-opensource-developer and thanks for reporting this issue! Apologies for the broken version.

Could you try cloning the master again, we're now using ipfs-daemon version `0.3.0-beta.3` which includes a fix for your test failures.

**Comment by wigy-opensource-developer** [comment]
Hi @haadcode! Did a fresh `git clone` from master commit 4131467. `npm install` then `npm test` resulted in the following error: `Error: Cannot find module 'ipfsd-ctl'`. The failing import is at `orbit-db/test/client.test.js:10:18`

I did an `npm install ipfsd-ctl`, but that only succeeds when an `ipfs daemon` is running on the development machine.

After that succeeded I did an `npm test` that failed with the following:

```
  Orbit Client with js-ipfs-api
    1) "before all" hook
2016-11-29T18:28:30.685Z [ERROR] ipfs-daemon: { TypeError: done is not a function
    at node_modules/ipfsd-ctl/lib/index.js:22:7
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9) uncaught: true }
(node:4869) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'stopDaemon' of undefined
    2) "after all" hook


  106 passing (21s)
  2 failing

  1) Orbit Client with js-ipfs-api "before all" hook:
     Uncaught TypeError: done is not a function
      at node_modules/ipfsd-ctl/lib/index.js:22:7
      at _combinedTickCallback (internal/process/next_tick.js:67:7)
      at process._tickDomainCallback (internal/process/next_tick.js:122:9)

  2) Orbit Client with js-ipfs-api "after all" hook:
     Error: Timeout of 20000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

I could not get to the same error that I reported before.

After all these experiences, I concluded that it is too early to rely on this project in an application prototype we were planning. I will check back in some months to see how it progresses, but I need to go on for now. Keep up the good work meanwhile.

**Comment by haadcode** [comment]
Hi @wigy-opensource-developer, sorry for your troubles :/ 

These problems are totally on me. You caught us on bad timing as we're preparing a major new release. I'm also sorry I didn't check my previous instructions: 0.15.2 is indeed the master while the fixed, new version (that I should've pointed you to) is still in a branch `0.16.0-beta1`. 

If you're willing to give it one more try, install orbit-db with `npm i orbit-db@0.16.1-beta.1` or clone it from that branch. The tests are running fine for that branch on my computer and in CI, so I believe you should be able to use it out of the box. Let me know if you end up doing this and if you still have troubles. If you don't have the time and will for now, please do check back in a week or so.

**Comment by haadcode** [comment]
This should be fixed now in master and in 0.16.0. Thanks for reporting the issue @wigy-opensource-developer!
